numpyro/contrib/indexing.py

```diff
@@ -0,0 +1,145 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import jax.numpy as np
+
+
+def _is_batched(arg):
+    return isinstance(arg, np.ndarray) and len(arg.shape) > 0
```

Comment:
nit: it is better to return `np.ndim(x) > 0` here, which also works for DeviceArray and Python scalars.